Util/Docker/carla-ue4.dockerfile

```diff
@@ -0,0 +1,168 @@
+# Set the XDG_RUNTIME_DIR environment variable for ./Setup.sh
+# ------------------------------------------------------------------------------
+ENV XDG_RUNTIME_DIR=/run/user/1000
+
+# ------------------------------------------------------------------------------
+# Explicitly tell Vulkan to use NVIDIA's ICD
+# ------------------------------------------------------------------------------
+ENV VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json
+
+# ------------------------------------------------------------------------------
+# Create or rename user/group "$USERNAME"
+# so that we end up with:
+# - group "$USERNAME" at GID=$GROUP_ID
+# - user "$USERNAME"  at UID=$USER_ID
+# ------------------------------------------------------------------------------
+RUN if [ -z "$(getent group $GROUP_ID)" ]; then \
+      groupadd -g $GROUP_ID "$USERNAME"; \
+    else \
+      existing_group="$(getent group $GROUP_ID | cut -d: -f1)"; \
+      if [ "$existing_group" != "$USERNAME" ]; then \
+        groupmod -n "$USERNAME" "$existing_group"; \
+      fi; \
+    fi && \
+    if [ -z "$(getent passwd $USER_ID)" ]; then \
+      useradd -m -u $USER_ID -g $GROUP_ID "$USERNAME"; \
+    else \
+      existing_user="$(getent passwd $USER_ID | cut -d: -f1)"; \
+      if [ "$existing_user" != "$USERNAME" ]; then \
+        usermod -l "$USERNAME" -d /home/"$USERNAME" -m "$existing_user"; \
+      fi; \
+    fi
+
+# ------------------------------------------------------------------------------
+# Install Unreal Engine build dependencies
+# ------------------------------------------------------------------------------
+RUN apt-get update && \
+    apt-get install -y \
+    git \
+    ca-certificates \
+    curl \
+    wget \
+    build-essential \
+    xdg-user-dirs \
+    xdg-utils \
+    sudo \
+    libvulkan1 \
+    mesa-vulkan-drivers \
+    libsdl2-dev \
+    vulkan-tools
+    # (optional) vulkan-tools -> useful for debugging
+
+# ------------------------------------------------------------------------------
+# Install CARLA build dependencies (make PythonAPI, make CarlaUE4Editor)
+# ------------------------------------------------------------------------------
+RUN apt-get install -y \
+    cmake \
+    ninja-build \
+    python3 \
+    python3-dev \
+    python3-setuptools \
+    python3-distro \
+    python3-wheel \
+    libtiff5-dev \
+    libjpeg-dev \
+    autoconf \
+    rsync \
+    unzip
+
+# ------------------------------------------------------------------------------
+# (Optional) Install CARLA build packaging depenencies(make build.utils)
+# ------------------------------------------------------------------------------
+RUN apt-get install -y \
+    libxml2-dev 
+
+# ------------------------------------------------------------------------------
+# Cleanup
+# ------------------------------------------------------------------------------
+RUN rm -rf /var/lib/apt/lists/*
```

Comment:
I believe this cleanup isn't taking effect due to how Docker layers work. Deleting files in a later layer doesn't reduce the final image size. The cleanup should be included in the same `RUN` instruction as the `apt update`. Resolving this for now to avoid blocking the PR.